pkg/init/cmd/service/runc.go

```diff
@@ -103,9 +136,33 @@ func runcInit(rootPath string) int {
 
 		cleanup(path)
 		_ = os.Remove(pidfile)
+
+		// dump the log file outputs to os.Stdout/os.Stderr
+		if _, err = stdout.Seek(0, 0); err != nil {
```

Comment:
At first I thought this `Seek` was actually doing something subtle/clever, like testing if `stdout` is a file or something else (a tty etc), but now I realise you just want to rewind it to dump it (the error message was a big clue and probably mitigates the need for a comment).

I've a vague unease about reusing the fd and in particular the requirement it adds to pass a readable stdout/err to the process (it could be `O_WRONLY` otherwise I think). I don't think that unease is particularly well founded given the process will have exited by this point so there's no buffering etc to worry about and I don't think there is actually anything inherently wrong with stdout/err being readable (although I've not consulted any of the specs).

Reply:
> doing something subtle/clever, like testing if stdout is a file or something else 

I think `runc` does something a little too clever like that, hence all of these headaches here. I am trying to avoid it. :-)

> I've a vague unease about reusing the fd and in particular the requirement it adds to pass a readable stdout/err to the process

I can see that. I cannot really close the fd since I have a `defer f.Close()` earlier, which needs to stay. Maybe it is cleaner to use a separate fd and dump that. I don't mind.